homeassistant/components/aladdin_connect/sensor.py

```diff
@@ -89,14 +98,17 @@ def __init__(
         device: DoorDevice,
         description: AccSensorEntityDescription,
     ) -> None:
-        """Initialize a sensor for an Abode device."""
+        """Initialize a sensor for an Aladdin Connect device."""
         self._device_id = device["device_id"]
         self._number = device["door_number"]
         self._name = device["name"]
+        self._model = device["model"]
         self._acc = acc
         self.entity_description = description
         self._attr_unique_id = f"{self._device_id}-{self._number}-{description.key}"
         self._attr_has_entity_name = True
+        if self._model == "01" and description.key != "rssi":
+            description.entity_registry_enabled_default = True
```

Comment:
Descriptions should not be dynamically amended. You should instead override the property, for example using the instance attributes:
```suggestion
            self._attr_entity_registry_enabled_default = True
```

Reply:
Thanks - I was struggling with the "correct" way to do this.